### DISCUSSION TITLE
How does TrustWallet generate public address and private key from mnemonic phrase ?

### DISCUSSION THREAD

**Discussion by NebelskyiDmytro** [trustwallet/wallet-core#3770] [Q&A]
When I create wallet in Trust Wallet and try to receive DOGECOIN, I can see 'X' address to receive it.
When I want to generate new random mnemonic phrase and public address and private key from this phrase using Python (for personal purposes), I could see another keys.

When I use my TrustWallet mnemonic in Python, it generate address than differs from TrustWallet, how does it work ? Can soneone explain the algorythm please ?

**Comment by satoshiotomakan** [comment]
Hi @NebelskyiDmytro, thank you for the question!
In order to generate same private key and therefore a DOGE address, you need to use the same [derivation path](https://river.com/learn/terms/d/derivation-path/). So the Python library you use may derive a different private key from the same mnemonic if the derivation path is different.
For example, in Trust Wallet we use `m/44'/3'/0'/0/0`.

**Comment by NebelskyiDmytro** [comment]
@satoshiotomakan thanks for your fast reply, sorry for double asking but is it possible to create 24 words mnemonic wallet in TW ? I want my wallet be more secured

**Reply by satoshiotomakan** [reply]
If I'm not mistaken, we do not allow to create 24 words mnemonic, but allow to import 12, 18 or 24 words mnemonics

**Comment by NebelskyiDmytro** [comment]
Is somewhere library for PYTHON, twhere I can use derivation path 'm/44'/3'/0'/0/0.' for DOGECOIN wallets ?

**Reply by satoshiotomakan** [reply]
I'm sorry, I can't answer this question
